src/AppWindow.vala

```diff
@@ -132,19 +103,43 @@ public abstract class AppWindow : PageWindow {
 
         ui.ensure_update ();
         add_accel_group (ui.get_accel_group ());
-
-        build_header_bar ();
     }
 
-    protected virtual void build_header_bar () {
+    construct {
+        assert (command_manager == null);
+        command_manager = new CommandManager ();
+        command_manager.altered.connect (on_command_manager_altered);
+
         redo_btn = new Gtk.Button.from_icon_name ("edit-redo", Gtk.IconSize.LARGE_TOOLBAR);
         redo_btn.action_name = ACTION_PREFIX + ACTION_REDO;
 
         undo_btn = new Gtk.Button.from_icon_name ("edit-undo", Gtk.IconSize.LARGE_TOOLBAR);
         undo_btn.action_name = ACTION_PREFIX + ACTION_UNDO;
 
+        header = new Gtk.HeaderBar ();
+        header.show_close_button = true;
         header.pack_end (redo_btn);
         header.pack_end (undo_btn);
+
+        icon_name = "multimedia-photo-manager";
+        title = _(Resources.APP_TITLE);
+
+        add_action_entries (action_entries, this);
+        set_titlebar (header);
+
+        Application.get_instance ().set_accels_for_action (ACTION_PREFIX + ACTION_FULLSCREEN, {"F11"});
```

Comment:
It'd be nice to also store the result of `Application.get_instance ()` as it is reused multiple times